src/node/desktop/src/main/detect-r.ts

```diff
@@ -55,34 +58,56 @@ function executeCommand(command: string): Expected<string> {
   }
 }
 
-export async function promptUserForR(): Promise<Expected<string | null>> {
-  // nothing to do if RSTUDIO_WHICH_R is set
-  const rstudioWhichR = getenv('RSTUDIO_WHICH_R');
-  if (rstudioWhichR) {
-    return ok(rstudioWhichR);
-  }
+export async function promptUserForR(platform = process.platform): Promise<Expected<string | null>> {
+  if (platform === 'win32') {
+    const desktop = await import('../native/desktop.node');
```

Comment:
Any reason to put the import here rather than at the top level?

Reply:
Well, when I was using an external node package for this feature, the package would require to only be installed on Windows OS. So I had to do it the import that way. So due to that, I think that doing it like this would probably save us from some headache that could or could not happen in the future as it won't hurt to import it like that, by the contrary: that's lazy import for this module which will only really import it when it's used...